### PR TITLE
E2E: actually fail when no InternalIP, ssh master tweaks, delete retries

### PR DESCRIPTION
**Pull request by jackfrancis**
**What this PR does / why we need it**: Prior usage was not failing when a node had no InternalIP. Additionally, simplify "the ssh to master" implementations, and retry all deletes.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**If applicable**:
- [ ] documentation
- [ ] unit tests
- [ ] tested backward compatibility (ie. deploy with previous version, upgrade with this branch)

**Release note**:

```
E2E: actually fail when no InternalIP
```

### CODE REVIEW

File: test/e2e/kubernetes/deployment/deployment.go [hunk 1]
```diff
@@ -134,22 +134,39 @@ func Get(name, namespace string) (*Deployment, error) {
...
+			}
+			break
 		}
 	}
+
+	if kubectlError != nil {
+		return kubectlError
+	}
+
 	return nil
```
Comment (CecileRobertMichon) [inline test/e2e/kubernetes/deployment/deployment.go]: can we just return kubectlError?

File: test/e2e/kubernetes/kubernetes_test.go [hunk 2]
```diff
@@ -389,16 +347,12 @@ var _ = Describe("Azure Container Cluster using the Kubernetes Orchestrator", fu
 							address := node.Status.GetAddressByType("InternalIP")
 							if address == nil {
 								log.Printf("One of our nodes does not have an InternalIP value!: %s\n", node.Metadata.Name)
+								Fail("All nodes should have an InternalIP address")
 							}
-							Expect(address).NotTo(Equal(nil))
```
Comment (CecileRobertMichon) [inline test/e2e/kubernetes/kubernetes_test.go]: was this not failing when address was nil?

File: test/e2e/kubernetes/deployment/deployment.go (hunk 1 above)
Comment (jackfrancis) [reply test/e2e/kubernetes/deployment/deployment.go]: yep :)

File: test/e2e/kubernetes/kubernetes_test.go [hunk 3]
```diff
@@ -30,14 +30,16 @@ import (
 )
 
 const (
-	WorkloadDir = "workloads"
-	PolicyDir   = "workloads/policies"
-	SSHPort     = "50001"
+	WorkloadDir           = "workloads"
+	PolicyDir             = "workloads/policies"
+	deleteResourceRetries = 10
```
Comment (CecileRobertMichon) [inline test/e2e/kubernetes/kubernetes_test.go]: if we are going to use a const for number of retries and it's the same for every call, why pass it as a parameter to the function and not just use it in the delete function directly?

File: test/e2e/kubernetes/persistentvolumeclaims/persistentvolumeclaims.go [hunk 4]
```diff
@@ -85,14 +85,23 @@ func Describe(pvcName, namespace string) error {
...
+			continue
+		}
+		break
+	}
+
+	if kubectlError != nil {
+		return kubectlError
 	}
+
 	return nil
```
Comment (CecileRobertMichon) [inline test/e2e/kubernetes/persistentvolumeclaims/persistentvolumeclaims.go]: same comment about returning kubectlError here

File: test/e2e/kubernetes/service/service.go [hunk 5]
```diff
@@ -74,14 +74,23 @@ func Get(name, namespace string) (*Service, error) {
...
+			continue
+		}
+		break
 	}
+
+	if kubectlError != nil {
+		return kubectlError
+	}
+
 	return nil
```
Comment (CecileRobertMichon) [inline test/e2e/kubernetes/service/service.go]: ditto

File: test/e2e/kubernetes/kubernetes_test.go (hunk 3 above)
Comment (jackfrancis) [reply test/e2e/kubernetes/kubernetes_test.go]: Good question. Just an iterative placeholder until we have stronger opinions about particular use cases of delete that should be retried fewer or more times (or until we make it user-configurable)

File: test/e2e/kubernetes/kubernetes_test.go (hunk 2 above)
Comment (jackfrancis) [reply test/e2e/kubernetes/kubernetes_test.go]: I think the way ginkgo implements the equality check doesn't work in the way it was being used.

File: test/e2e/kubernetes/kubernetes_test.go (hunk 3 above)
Comment (tariq1890) [reply test/e2e/kubernetes/kubernetes_test.go]: Thoughts on having a generic returnable function that takes in function/cmd parameters?

File: test/e2e/kubernetes/kubernetes_test.go (hunk 2 above)
Comment (CecileRobertMichon) [reply test/e2e/kubernetes/kubernetes_test.go]: looking at gingko docs it looks like `Fail()`  won't run the `AfterSuite()` stuff so maybe not ideal

http://onsi.github.io/ginkgo/#getting-started-writing-your-first-test

Comment (CecileRobertMichon) [reply test/e2e/kubernetes/kubernetes_test.go]: could we maybe add an `else{}` to protect against panic instead and let the assertion do its thing?

Comment (jackfrancis) [reply test/e2e/kubernetes/kubernetes_test.go]: Good catch. What do you think about this `getAddressError` implementation?

Comment (CecileRobertMichon) [reply test/e2e/kubernetes/kubernetes_test.go]: I don't think that will help. The succeeding line will still be run without an else statement to protect it. I think `Expect(address).NotTo(Equal(nil))` <==> `Expect(getAddressError).NotTo(HaveOccurred())`

Comment (CecileRobertMichon) [reply test/e2e/kubernetes/kubernetes_test.go]: Should be `Expect(address).NotTo(BeNil())`

**Pull request merged**